diff -Naur Linux-PAM-0.99.10.0/doc/specs/Makefile.in Linux-PAM-0.99.10.0/doc/specs/Makefile.in
--- Linux-PAM-0.99.10.0/doc/specs/Makefile.in	2008-02-13 20:52:54.000000000 +0700
+++ Linux-PAM-0.99.10.0/doc/specs/Makefile.in	2015-11-07 13:52:31.303228000 +0700
@@ -555,7 +555,7 @@
 
 
 draft-morgan-pam-current.txt: padout draft-morgan-pam.raw
-	./padout < $(srcdir)/draft-morgan-pam.raw > draft-morgan-pam-current.txt
+	#./padout < $(srcdir)/draft-morgan-pam.raw > draft-morgan-pam-current.txt
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Naur Linux-PAM-0.99.10.0/modules/pam_unix/pam_unix_acct.c Linux-PAM-0.99.10.0/modules/pam_unix/pam_unix_acct.c
--- Linux-PAM-0.99.10.0/modules/pam_unix/pam_unix_acct.c	2008-01-28 19:21:48.000000000 +0700
+++ Linux-PAM-0.99.10.0/modules/pam_unix/pam_unix_acct.c	2015-11-07 11:04:23.865565000 +0700
@@ -48,6 +48,10 @@
 #include <errno.h>
 #include <sys/wait.h>
 
+#include <sys/types.h> 
+#include <sys/time.h> 
+#include <sys/resource.h>
+
 #include <security/_pam_macros.h>
 
 /* indicate that the following groups are defined */
