diff -Naur udev-lfs-208-1/Makefile.lfs udev-lfs-208-1/Makefile.lfs
--- udev-lfs-208-1/Makefile.lfs	2013-10-02 23:47:46.000000000 +0700
+++ udev-lfs-208-1/Makefile.lfs	2015-11-06 09:57:29.340717000 +0700
@@ -29,7 +29,7 @@
 OPTIONS2 := $(OPTIONS) -fvisibility=hidden
 
 LDFLAGS1  = -pthread -lrt -Wl,--as-needed -Wl,--gc-sections -Wl,--no-undefined 
-LDFLAGS2 := $(LDFLAGS1) -lblkid -lkmod 
+LDFLAGS2 := $(LDFLAGS) $(LDFLAGS1) -lblkid -lkmod -luuid
 
 DEF = -include cfg.h                    \
       -DSYSCONFDIR=\"/etc\"             \
@@ -41,7 +41,7 @@
       -D_FILE_OFFSET_BITS=64            \
       -DHAVE_FIRMWARE
 
-INCLUDE = -I src/libudev -I src/shared -I src -I src/login -I src/systemd \
+INCLUDE = $(CFLAGS) -I src/libudev -I src/shared -I src -I src/login -I src/systemd \
           -I src/core    -I udev-lfs-$(SYSTEMD_VERSION)-1
 
 LIBUDEV_SRCS = libudev-device-private.c \
